lib/models/Thread.js

```diff
@@ -47,6 +47,88 @@ export default class Thread extends Model {
   }
 
   /**
+   * Updates the isFlagged field of all {@link Message}s of this {@link Thread}.
+   *
+   * @param isFlagged {Boolean} The isFlagged field of the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  setIsFlagged(isFlagged) {
+    return this._flagAllMessages('isFlagged', isFlagged);
+  }
+
+  /**
+   * Updates the isUnread field of this {@link Message}.
+   *
+   * @param isUnread {Boolean} The isUnread field of the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  setIsUnread(isUnread) {
+    return this._flagAllMessages('isUnread', isUnread);
+  }
+
+  /**
+   * Moves this {@link Thread} to a different set of mailboxes.
+   *
+   * @param mailboxIds {String[]} The identifiers of the target mailboxes for the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  move(mailboxIds) {
+    return this._updateAllMessages({ mailboxIds });
+  }
+
+  /**
+   * Moves this {@link Thread} to the mailbox holding the given `role`.<br />
+   * This will first do a JMAP request to find the mailbox with the given role, then a {@link Thread#move} to move the message.
+   *
+   * @param role {MailboxRole|String} The desired mailbox role.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Thread#move}.
+   *
+   * @see MailboxRole
+   * @see Client#getMailboxWithRole
+   */
+  moveToMailboxWithRole(role) {
+    return this._jmap.getMailboxWithRole(role).then(mailbox => this.move([mailbox.id]));
+  }
+
+  /**
+   * Destroy this {@link Thread} on the server.<br />
+   * This will internally destory all messages contained in this thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.destroyMessages}.
+   *
+   * @see Client#destroyMessages
+   */
+  destroy() {
+    return this._jmap.destroyMessages(this.messageIds);
+  }
+
+  _updateAllMessages(updates) {
+    let options = {};
+
+    this.messageIds.forEach(id => {
+      options[id] = updates;
+    });
+
+    return this._jmap.setMessages({ update: options });
+  }
+
+  _flagAllMessages(flag, state) {
+    Utils.assertRequiredParameterHasType(state, flag, 'boolean');
```

Comment:
Il fait si mauvais que ça du côté de poissy ? 😉

Envoyé avec Boxer couplé à OBM

Le 31 janv. 2016 11:41, ddolcimascolo <notifications@github.com> a écrit :

In lib/models/Thread.js:

> + destroy() { > + return this._jmap.destroyMessages(this.messageIds); > + } > + > + _updateAllMessages(updates) { > + let options = {}; > + > + this.messageIds.forEach(id => { > + options[id] = updates; > + }); > + > + return this._jmap.setMessages({ update: options }); > + } > + > + _flagAllMessages(flag, state) { > + Utils.assertRequiredParameterHasType(state, flag, 'boolean'); 

Ok for state, not for flag. This in an internal method and flag is passed by other methods directly, not by client code.

—
Reply to this email directly or view it on GitHub.